### PR TITLE
fix: correct five merge-engine edge cases found in a functional audit

### PR DESCRIPTION
**Pull request by yohanim**
# Explain your changes

---

## Summary

A 73-case functional audit of the three-way merge engine surfaced several
edge cases outside the existing test suite's coverage.

- **fix(merger)**: `UnkeyedConflictStrategy` no longer conflicts an unkeyed
  repeated-element array that's actually unchanged.
- **fix(writer)**: two compounding layout bugs that glued conflict-marker
  lines together whenever a conflict side was contentless.
- **fix(driver)**: internal merge failures (e.g. malformed XML) are now
  logged and rethrown instead of being swallowed into `hasConflict=true`,
  so `bin/driver.ts` reports them and exits 1. `ours` is still left
  untouched on disk.
- **fix(merger)**: root `xmlns` merge now resolves three-way (unchanged-
  on-one-side defers to the other side's change) instead of always
  letting `other` win, even when only `local` had changed. Only the
  genuine all-three-different tie — which can't carry a zdiff3 marker
  inside an XML attribute — still has to pick a side; it now keeps
  `local` and logs the discarded value instead of silently dropping it.
- **fix(merger)**: `package.xml` `<members>` (and any future attribute
  listed in `MetadataService.isTextArrayAttribute`) now gets consistent
  set-semantics merging regardless of how many occurrences happen to
  exist — previously a `<types>` block with exactly one `<members>`
  entry conflicted on a rename that a block with two or more would have
  silently resolved. The override stays scoped to scalar values — an
  element of the same name carrying children keeps merging property by
  property.
- **docs(design)**: Known Limitations updated to match — two entries
  from an earlier pass turned out to be real bugs on closer inspection
  (their "this is intentional" evidence didn't hold up), not design
  decisions, so they're fixed rather than documented.

## Test plan

- [x] New/updated unit tests for every fix (each verified to fail
      without the fix and pass with it via `git stash`)
- [x] Three new fixtures exercising the full parse→merge→write pipeline:
      `51-unkeyed-array-unchanged-sibling-change`,
      `52-conflict-whole-file-emptied-vs-modified`,
      `53-package-xml-members-cardinality-one-rename`
- [x] Extended integration coverage for the rethrow
      (`test/integration/MergeDriver.test.ts`)
- [x] `npm test` green (compile, lint, unit, integration, functional, nut)
- [x] Coverage thresholds (100% branches/functions/lines/statements) met

# Does this close any currently open issues?

---

closes #

- [ ] Jest tests added to cover the fix.
- [ ] NUT tests added to cover the fix.
- [ ] E2E tests added to cover the fix.

# Any particular element that can be tested locally

---

# Any other comments

---

### CODE REVIEW

File: src/driver/MergeDriver.ts [hunk 1]
```diff
@@ -92,6 +92,15 @@ export class MergeDriver {
...
       }
       Logger.error('Merge failed; leaving ours unchanged', error)
+      // The exit code alone can't distinguish this from a real conflict
+      // (both return hasConflict=true → exit 1), and Logger only reaches
+      // stderr when SF_LOG_STDERR is set — so without this, a git merge
+      // hitting this path leaves `ours` untouched with zero markers and
+      // zero visible indication anything failed. Surface it unconditionally.
+      const message = error instanceof Error ? error.message : String(error)
+      process.stderr.write(
+        `sf-git-merge-driver: merge failed for ${oursPath}; left unchanged: ${message}\n`
```
Comment (scolladon) [inline src/driver/MergeDriver.ts]: **issue (blocking):** ❗ it should use a message from our message layer

but doing so will requires MergeDriver to know about this whereas it should not
so it should propagate the issue for the plugin to do that write and to do it using message template

**Review by scolladon — COMMENTED** [review]
Let's discuss error propagation

File: src/driver/MergeDriver.ts (hunk 1 above)
Comment (scolladon) [reply src/driver/MergeDriver.ts]: Amending my own comment above, scratch the "use the message layer" part, that was
the wrong call.

Routing this through `Messages.loadMessages` would mean pulling `@salesforce/core`
into `bin/merge-driver.cjs`, which is the path git actually invokes. That bundle is
36 KB against a 300 KB gate; `@salesforce/core` is 2.3 MB before its transitives,
and `importMessagesDirectoryFromMetaUrl` needs `import.meta.url` plus a runtime
`messages/` directory, neither survives a CJS single-file bundle. Totally
unnecessary weight for one diagnostic line. The binary keeps its own literals
(there are already four in `src/bin/driver.ts`), that's fine and deliberate.

The layering half still stands though, so still blocking: `MergeDriver` shouldn't
touch `process.stderr`. Every other stderr write in the codebase lives in
`src/bin/driver.ts`, and the test fallout shows the coupling — `MergeDriver.test.ts`
now needs a blanket `vi.spyOn(process.stderr, 'write')` in `beforeEach` just to keep
unrelated tests quiet.

The simpler fix is to drop the block and rethrow:

```ts
Logger.error('Merge failed; leaving ours unchanged', error)
throw error
```

`src/bin/driver.ts:140-152` already catches, writes
`sf-git-merge-driver: <message>` and returns `CONFLICT_EXIT_CODE`. `ours` stays
untouched because the rename never ran, and the finally still cleans the tmp
file, same observable behavior for git, with a smaller diff and the stderr spies
gone from the tests.

The underlying point: swallowing the error and returning true isn't domain logic.
Mapping "merge failed" onto git's conflict exit code is a presentation decision, and
it belongs at the same boundary that already owns the message. Rethrowing moves both
there in one step.

Two details worth handling:
- Wrap the error on the way out if you want oursPath in the output, rather than
teaching the bin about paths.
- The deprecated run command would let the throw reach SfCommand, shifting its
exit code 1 → 2. Git reads any non-zero as a conflict so it's cosmetic, but call
it out in the PR description.

**Review by scolladon — APPROVED** [review]
LGTM

Great work, better harness, good intent tracking -> project is more robust
